File: 2038-Remove_Colored_Pieces_if_Both_Neighbors_are_the_Same_Color/Remove_Colored_Pieces_if_Both_Neighbors_are_the_Same_Color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        if(colors.length() < 3) return false;
        
        int[] removableColors = new int[2];
        
        int count = 0;
        char currColor = colors.charAt(0);
        
        for(int i = 0; i < colors.length(); i++){
            if(currColor == colors.charAt(i)){
                count++;
            } else {
                if(count > 2) removableColors[currColor - 'A'] += (count - 2);
                currColor = colors.charAt(i);
                count = 1;
            }
        }
        
        if(count > 2) removableColors[currColor - 'A'] += (count - 2);
        
        
        return removableColors[0] > removableColors[1];
    }
}

File: 2038-RemoveColoredPiecesIfBothNeighborsAreTheSameColor/RemoveColoredPiecesIfBothNeighborsAreTheSameColor.go
package removecoloredpiecesifbothneighborsarethesamecolor

func winnerOfGame(colors string) bool {
    if len(colors) < 3 {
        return false
    }
    
    
    removableColors := make([]int, 2)
    
    count := 0
    currColor := colors[0]
    
    for i := 0; i < len(colors); i++ {
        if colors[i] == currColor {
            count++
        } else {
            if count > 2 {
                removableColors[currColor - 'A'] += (count - 2)
            }
            count = 1
            currColor = colors[i]
        }
    }
    
    if count > 2 {
        removableColors[currColor - 'A'] += (count - 2)
    }
    
    return removableColors[0] > removableColors[1]
     
}